tests/src/test/java/org/eclipse/hono/tests/http/HttpTestBase.java

```diff
@@ -920,6 +922,39 @@ public void testUploadMessagesWithTtdThatReplyWithCommand(
             final VertxTestContext ctx) throws InterruptedException {
 
         final Tenant tenant = new Tenant();
+        testUploadMessagesWithTtdThatReplyWithCommand(endpointConfig, tenant, ctx);
+    }
+
+    /**
+     * Verifies that the HTTP adapter delivers a command to a device and accepts the corresponding
+     * response from the device, all while using a tenant that has the "support-concurrent-gateway-device-command-requests"
+     * option set.
+     *
+     * @param endpointConfig The endpoints to use for sending/receiving commands.
+     * @param ctx The test context.
+     * @throws InterruptedException if the test fails.
+     */
+    @ParameterizedTest(name = IntegrationTestSupport.PARAMETERIZED_TEST_NAME_PATTERN)
+    @MethodSource("commandAndControlVariants")
+    public void testUploadMessagesWithTtdThatReplyWithCommandWithGatewayMappingDisabled(
+            final HttpCommandEndpointConfiguration endpointConfig,
+            final VertxTestContext ctx) throws InterruptedException {
+
+        if (endpointConfig.isSubscribeAsGatewayForAllDevices()) {
+            logger.info("test skipped (GATEWAY_FOR_ALL_DEVICES not supported if gateway mapping disabled)");
+            ctx.completeNow();
+            return;
+        }
```

Comment:
This looks like a hack to me ;-) IMHO we should simply have another method source for the parameters that this test supports ...

Reply:
Right, I've committed a cleaner solution.